test/cpp_extensions/open_registration_extension.cpp

```diff
@@ -23,13 +26,50 @@ C10_REGISTER_GUARD_IMPL(PrivateUse1, c10::impl::NoOpDeviceGuardImpl<DeviceType::
 
 }} // namespace at::detail
 
+namespace {
+
+template <typename T>
+static T abs_impl(T v) {
+  return std::abs(v);
+}
+
+template <>
+C10_UNUSED uint8_t abs_impl(uint8_t v) {
+  return v;
+}
+
+void abs_kernel(TensorIteratorBase& iter) {
+  auto dtype = iter.dtype();
+  if (dtype == kComplexHalf) {
+    using scalar_t = c10::complex<Half>;
+    using opmath_t = at::opmath_type<scalar_t>;
+    cpu_kernel(iter, [=](scalar_t a) -> scalar_t { return abs_impl(opmath_t{a}); });
+  } else {
+    AT_DISPATCH_ALL_TYPES_AND_COMPLEX_AND2(kBFloat16, kHalf, iter.dtype(), "abs_cpu", [&]() {
+      cpu_kernel_vec(
+          iter,
+          [=](scalar_t a) -> scalar_t { return abs_impl(a); },
+          [=](Vectorized<scalar_t> a) { return a.abs(); });
+    });
+  }
+}
+
+REGISTER_PRIVATEUSE1_DISPATCH(abs_stub, &abs_kernel);
```

Comment:
it would be nice to be able to test that we actually ran this kernel (right now the "test" just calls `abs()` and checks that the outputs match).

One way to do it would be the way it's done in `custom_add_Tensor` below - add a global counter for whether our custom abs kernel was called, that we increment inside of the kernel. Then expose a `custom_abs_called` helper to python that you can assert in the python code.

Reply:
Thank you for the suggestion. I have modified the unit test according to your suggestion, added `custom_abs_called` and `abs_counter` in test.